### PR TITLE
bastion越しでdeployできるようにする

### PR DESCRIPTION
**Pull request by ogidow**

### CODE REVIEW

**Comment by ogidow** [comment]
### デプロイ

<details>
<summary>

$ cap production deploy</summary>



00:00 git:wrapper
      01 mkdir -p /tmp/railstutorial/
    ✔ 01 deploy@all 0.459s
      Uploading /tmp/railstutorial/git-ssh.sh 100.0%
      02 chmod +rx /tmp/railstutorial/git-ssh.sh
    ✔ 02 deploy@all 0.007s
00:00 git:check
      01 git ls-remote --heads https://github.com/ogidow/railstutorial.git
      01 97a0cadb766e1468aae5ed4aef03bae07983d659   refs/heads/api-update-destroy
      01 b3bb74bbb4e4b904e49186dcd0885675efdbf48c   refs/heads/chapter1
      01 3b53a2ba027a235c59ff871de5506311b8e3c188   refs/heads/chapter10
      01 e992aa6af99ed94b24c0c4f0d3e5310ff10d3a00   refs/heads/chapter10-ex
      01 de051a1db07aaddb2def426d280edfbd8a34e6bb   refs/heads/chapter11
      01 0915d79cc439c422a5031ab304fb1e0383d20fd1   refs/heads/chapter11-1
      01 35a9739ed6f4d6ed0797a9dcade9fa093ab8fbfd   refs/heads/chapter11-2
      01 cfa1af7cedf1d5175894347445212ccd32635199   refs/heads/chapter11-3
      01 59f7aceb21bd0cd3c002a2da4ba71e1a0d3a3677   refs/heads/chapter11-4
      01 8c92e0978892b4f2972b6308e14644aa6cd0a66a   refs/heads/chapter12-2
      01 b0a88192d0eaaddf5907025818428af637f1ba6d   refs/heads/chapter12-3
      01 56ab066318808e46337b0c7a4265fe34408f3676   refs/heads/chapter12-ex
      01 b890d7cd53bd932290e60480ec9aceb4995d099a   refs/heads/chapter7
      01 1f58fbd9944615799add7f496b219b7786b67f59   refs/heads/chapter7-ex
      01 3e8cb57c7c9da630786227ced88f9e3a4955e9c2   refs/heads/chapter8
      01 0c1fdc60320658bef827d6eeb2d0515c2ae4e320   refs/heads/chapter8-ex
      01 48f82f842c960c5619b9b82cac73cb7ea100f6b1   refs/heads/chapter9
      01 701e7b0ff197d931895b8dde2e45c26eadcc77ed   refs/heads/chapter9-ex
      01 b40d809272d3351f323591fc7361ff31a82abe20   refs/heads/deploy-via-bastion
      01 459b3376891955fa44abee2a16e5173170d57c1b   refs/heads/follow-api
      01 f6999eb2a6a1fc8ea5efe659ace94c231a54308a   refs/heads/master
      01 22dc251aaa3a0f99cbe96d0f5f531b35bbc68c0b   refs/heads/microposts-api
      01 6d1287e9ebf01b51adbf3a7b24397e85b6f1cf8d   refs/heads/travis-ci
    ✔ 01 deploy@all 1.803s
00:02 deploy:check:directories
      01 mkdir -p /var/www/railstutorial/shared /var/www/railstutorial/releases
    ✔ 01 deploy@all 0.006s
00:02 deploy:check:linked_dirs
      01 mkdir -p /var/www/railstutorial/shared/log /var/www/railstutorial/shared/tmp/pids /var/www/railstutorial/shared/tmp/cache /var/www/railstutorial/shared/tmp/sockets /var/ww…
    ✔ 01 deploy@all 0.007s
00:02 git:clone
      01 git clone --mirror https://github.com/ogidow/railstutorial.git /var/www/railstutorial/repo
      01 Cloning into bare repository '/var/www/railstutorial/repo'...
    ✔ 01 deploy@all 2.711s
00:05 git:update
      01 git remote update --prune
      01 Fetching origin
    ✔ 01 deploy@all 1.558s
00:07 git:create_release
      01 mkdir -p /var/www/railstutorial/releases/20160808040147
    ✔ 01 deploy@all 0.006s
      02 git archive master | tar -x -f - -C /var/www/railstutorial/releases/20160808040147
    ✔ 02 deploy@all 0.025s
00:07 git:set_current_revision
      01 echo "f6999eb2a6a1fc8ea5efe659ace94c231a54308a" >> REVISION
    ✔ 01 deploy@all 0.014s
00:07 deploy:symlink:linked_dirs
      01 mkdir -p /var/www/railstutorial/releases/20160808040147 /var/www/railstutorial/releases/20160808040147/tmp /var/www/railstutorial/releases/20160808040147/public
    ✔ 01 deploy@all 0.012s
      02 rm -rf /var/www/railstutorial/releases/20160808040147/log
    ✔ 02 deploy@all 0.010s
      03 ln -s /var/www/railstutorial/shared/log /var/www/railstutorial/releases/20160808040147/log
    ✔ 03 deploy@all 0.006s
      04 ln -s /var/www/railstutorial/shared/tmp/pids /var/www/railstutorial/releases/20160808040147/tmp/pids
    ✔ 04 deploy@all 0.009s
      05 ln -s /var/www/railstutorial/shared/tmp/cache /var/www/railstutorial/releases/20160808040147/tmp/cache
    ✔ 05 deploy@all 0.008s
      06 ln -s /var/www/railstutorial/shared/tmp/sockets /var/www/railstutorial/releases/20160808040147/tmp/sockets
    ✔ 06 deploy@all 0.013s
      07 ln -s /var/www/railstutorial/shared/public/assets /var/www/railstutorial/releases/20160808040147/public/assets
    ✔ 07 deploy@all 0.007s
00:07 bundler:install
      01 bundle install --path /var/www/railstutorial/shared/bundle --without development test --deployment --quiet
    ✔ 01 deploy@all 156.680s
02:44 deploy:assets:precompile
      01 bundle exec rake assets:precompile
      01 I, [2016-08-08T04:04:31.800541 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/rails-c094bc3a4bf50e5bb477109e5cb0d213af27ad75b481c4…
      01 I, [2016-08-08T04:04:35.495780 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/application-336e308138393c5a35e22e413531386023049c17…
      01 I, [2016-08-08T04:04:35.496677 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/application-336e308138393c5a35e22e413531386023049c17…
      01 I, [2016-08-08T04:04:39.452992 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/application-70b07e9b00684e21b9d1140c5f3e7a492d997237…
      01 I, [2016-08-08T04:04:39.453178 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/application-70b07e9b00684e21b9d1140c5f3e7a492d997237…
      01 I, [2016-08-08T04:04:39.457861 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-f495f34e4f177…
      01 I, [2016-08-08T04:04:39.458674 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-f495f34e4f177…
      01 I, [2016-08-08T04:04:39.459223 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-fc969dc1c6ff5…
      01 I, [2016-08-08T04:04:39.462557 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-bd18efd3efd70…
      01 I, [2016-08-08T04:04:39.463690 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-bd18efd3efd70…
      01 I, [2016-08-08T04:04:39.464576 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-d168d50a88c73…
      01 I, [2016-08-08T04:04:39.479400 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-d168d50a88c73…
      01 I, [2016-08-08T04:04:39.499453 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-d168d50a88c73…
      01 I, [2016-08-08T04:04:39.499903 #16947]  INFO -- : Writing /var/www/railstutorial/releases/20160808040147/public/assets/bootstrap/glyphicons-halflings-regular-d168d50a88c73…
    ✔ 01 deploy@all 10.382s
02:54 deploy:assets:backup_manifest
      01 mkdir -p /var/www/railstutorial/releases/20160808040147/assets_manifest_backup
    ✔ 01 deploy@all 0.006s
      02 cp /var/www/railstutorial/releases/20160808040147/public/assets/.sprockets-manifest-9ac44f0650ff0b310e36fa6a6c416ac6.json /var/www/railstutorial/releases/20160808040147/as…
    ✔ 02 deploy@all 0.007s
02:54 deploy:db:create
      01 bundle exec rake db:create
    ✔ 01 deploy@all 2.004s
02:56 deploy:migrate
      [deploy:migrate] Run `rake db:migrate`
02:56 deploy:migrating
      01 bundle exec rake db:migrate
      01    (34.4ms)  CREATE TABLE `schema_migrations` (`version` varchar(255) NOT NULL) ENGINE=InnoDB
      01    (37.2ms)  CREATE UNIQUE INDEX `unique_schema_migrations`  ON `schema_migrations` (`version`)
      01   ActiveRecord::SchemaMigration Load (0.2ms)  SELECT `schema_migrations`.\* FROM `schema_migrations`
      01 Migrating to CreateUsers (20160617071229)
      01 == 20160617071229 CreateUsers: migrating ======================================
      01
      01 -- create_table(:users)
      01
      01    (32.9ms)  CREATE TABLE `users` (`id` int(11) auto_increment PRIMARY KEY, `name` varchar(255), `email` varchar(255), `created_at` datetime NOT NULL, `updated_at` datetim…
      01    -> 0.0339s
      01
      01 == 20160617071229 CreateUsers: migrated (0.0344s) =============================
      01
      01
      01    (0.4ms)  BEGIN
      01   SQL (1.0ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160617071229')
      01    (2.9ms)  COMMIT
      01 Migrating to AddIndexToUsersEmail (20160620061703)
      01 == 20160620061703 AddIndexToUsersEmail: migrating =============================
      01
      01 -- add_index(:users, :email, {:unique=>true})
      01
      01    (25.6ms)  CREATE UNIQUE INDEX `index_users_on_email`  ON `users` (`email`)
      01    -> 0.0279s
      01
      01 == 20160620061703 AddIndexToUsersEmail: migrated (0.0288s) ====================
      01
      01
      01    (0.4ms)  BEGIN
      01   SQL (0.3ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160620061703')
      01    (2.1ms)  COMMIT
      01 Migrating to AddPasswordDigestToUsers (20160620064055)
      01 == 20160620064055 AddPasswordDigestToUsers: migrating =========================
      01
      01 -- add_column(:users, :password_digest, :string)
      01
      01    (40.0ms)  ALTER TABLE `users` ADD `password_digest` varchar(255)
      01    -> 0.0407s
      01
      01 == 20160620064055 AddPasswordDigestToUsers: migrated (0.0415s) ================
      01
      01
      01    (0.2ms)  BEGIN
      01   SQL (0.4ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160620064055')
      01    (10.9ms)  COMMIT
      01 Migrating to AddRememberDigestToUsers (20160624025342)
      01 == 20160624025342 AddRememberDigestToUsers: migrating =========================
      01
      01 -- add_column(:users, :remember_digest, :string)
      01
      01    (32.2ms)  ALTER TABLE `users` ADD `remember_digest` varchar(255)
      01    -> 0.0329s
      01
      01 == 20160624025342 AddRememberDigestToUsers: migrated (0.0338s) ================
      01
      01
      01    (0.2ms)  BEGIN
      01   SQL (0.4ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160624025342')
      01    (9.3ms)  COMMIT
      01 Migrating to AddAdminToUsers (20160627083741)
      01 == 20160627083741 AddAdminToUsers: migrating ==================================
      01
      01 -- add_column(:users, :admin, :boolean, {:default=>false})
      01
      01    (32.9ms)  ALTER TABLE `users` ADD `admin` tinyint(1) DEFAULT 0
      01    -> 0.0337s
      01
      01 == 20160627083741 AddAdminToUsers: migrated (0.0346s) =========================
      01
      01
      01    (0.2ms)  BEGIN
      01   SQL (0.3ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160627083741')
      01    (11.0ms)  COMMIT
      01 Migrating to AddActivationToUsers (20160629032630)
      01 == 20160629032630 AddActivationToUsers: migrating =============================
      01
      01 -- add_column(:users, :activation_digest, :string)
      01
      01    (21.0ms)  ALTER TABLE `users` ADD `activation_digest` varchar(255)
      01    -> 0.0214s
      01 -- add_column(:users, :activated, :boolean, {:default=>false})
      01    (37.2ms)  ALTER TABLE `users` ADD `activated` tinyint(1) DEFAULT 0
      01    -> 0.0377s
      01 -- add_column(:users, :activated_at, :datetime)
      01    (25.8ms)  ALTER TABLE `users` ADD `activated_at` datetime
      01    -> 0.0263s
      01 == 20160629032630 AddActivationToUsers: migrated (0.0858s) ====================
      01
      01    (0.3ms)  BEGIN
      01   SQL (0.6ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160629032630')
      01    (4.5ms)  COMMIT
      01 Migrating to AddResetToUsers (20160629074301)
      01 == 20160629074301 AddResetToUsers: migrating ==================================
      01 -- add_column(:users, :reset_digest, :string)
      01    (33.5ms)  ALTER TABLE `users` ADD `reset_digest` varchar(255)
      01    -> 0.0343s
      01 -- add_column(:users, :reset_sent_at, :datetime)
      01    (36.7ms)  ALTER TABLE `users` ADD `reset_sent_at` datetime
      01    -> 0.0373s
      01 == 20160629074301 AddResetToUsers: migrated (0.0718s) =========================
      01
      01    (0.4ms)  BEGIN
      01   SQL (0.6ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160629074301')
      01    (4.2ms)  COMMIT
      01 Migrating to CreateMicroposts (20160701065841)
      01 == 20160701065841 CreateMicroposts: migrating =================================
      01 -- create_table(:microposts)
      01    (18.7ms)  CREATE TABLE `microposts` (`id` int(11) auto_increment PRIMARY KEY, `content` text, `user_id` int(11), `created_at` datetime NOT NULL, `updated_at` datetime N…
      01    (37.9ms)  ALTER TABLE `microposts` ADD CONSTRAINT `fk_rails_558c81314b`
      01 FOREIGN KEY (`user_id`)
      01   REFERENCES `users` (`id`)
      01
      01    -> 0.0601s
      01 -- add_index(:microposts, [:user_id, :created_at])
      01    (18.9ms)  CREATE  INDEX `index_microposts_on_user_id_and_created_at`  ON `microposts` (`user_id`, `created_at`)
      01    -> 0.0211s
      01 == 20160701065841 CreateMicroposts: migrated (0.0814s) ========================
      01
      01    (1.6ms)  BEGIN
      01   SQL (0.8ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160701065841')
      01    (1.5ms)  COMMIT
      01 Migrating to AddPictureToMicroposts (20160705053705)
      01 == 20160705053705 AddPictureToMicroposts: migrating ===========================
      01
      01 -- add_column(:microposts, :picture, :string)
      01
      01    (31.2ms)  ALTER TABLE `microposts` ADD `picture` varchar(255)
      01    -> 0.0321s
      01 == 20160705053705 AddPictureToMicroposts: migrated (0.0326s) ==================
      01
      01    (0.3ms)  BEGIN
      01   SQL (1.2ms)  INSERT INTO `schema_migrations` (`version`) VALUES ('20160705053705')
      01    (4.0ms)  COMMIT
      01 Migrating to CreateRelationships (20160706051244)
      01 == 20160706051244 CreateRelationships: migrating ==============================
      01
      01 -- create_table(:relationships)
      01
      01    (39.8ms)  CREATE TABLE `relationships` (`id` int(11) auto_increment PRIMARY KEY, `follower_id` int(11), `followed_id` int(11), `created_at` datetime NOT NULL, `updated_…
      01    -> 0.0420s
      01
      01 -- add_index(:relationships, :follower_id)
      01
      01    (29.4ms)  CREATE  INDEX`index_relationships_on_follower_id`ON`relationships`(`follower_id`)
      01    -> 0.0340s
      01 -- add_index(:relationships, :followed_id)
      01    (20.5ms)  CREATE  INDEX`index_relationships_on_followed_id`ON`relationships`(`followed_id`)
      01    -> 0.0234s
      01 -- add_index(:relationships, [:follower_id, :followed_id], {:unique=>true})
      01    (15.6ms)  CREATE UNIQUE INDEX`index_relationships_on_follower_id_and_followed_id`ON`relationships`(`follower_id`,`followed_id`)
      01    -> 0.0171s
      01
      01 == 20160706051244 CreateRelationships: migrated (0.1209s) =====================
      01
      01
      01    (0.4ms)  BEGIN
      01   SQL (7.2ms)  INSERT INTO`schema_migrations`(`version`) VALUES ('20160706051244')
      01    (1.5ms)  COMMIT
    ✔ 01 deploy@all 3.047s
02:59 deploy:symlink:release
      01 ln -s /var/www/railstutorial/releases/20160808040147 /var/www/railstutorial/releases/current
    ✔ 01 deploy@all 0.008s
      02 mv /var/www/railstutorial/releases/current /var/www/railstutorial
    ✔ 02 deploy@all 0.006s
02:59 deploy:log_revision
      01 echo "Branch master (at f6999eb2a6a1fc8ea5efe659ace94c231a54308a) deployed as release 20160808040147 by usr0600438" >> /var/www/railstutorial/revisions.log
    ✔ 01 deploy@all

</details>

### ユニコーンの起動

<details>
<summary>

$ cap production unicorn:start</summary>


00:00 unicorn:start
      01 bundle exec unicorn -c /var/www/railstutorial/current/config/unicorn/production.rb -E deployment -D
    ✔ 01 deploy@all
</details>

File: config/deploy/production.rb [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
   user: 'deploy',
   roles: %w{app db web},
-  port: '2222',
   ssh_options: {
     keys: %w(~/.ssh/deploy_key),
-    forward_agent: false,
-    auth_methods: %w(publickey)
+    forward_agent: true,
+    auth_methods: %w(publickey),
+    proxy: Net::SSH::Proxy::Command.new('ssh -i ~/.ssh/bastion_key bastion@127.0.0.1 -p 2222 nc -w 2 %h %p')
```
Comment (ry023) [inline config/deploy/production.rb:17]: ここのipはいつか変わるやろうな〜
変えるときに忘れないようにせんとね

**Comment by ry023** [comment]
lgtm!

File: config/deploy/production.rb (hunk 1 above)
Comment (shimoju) [inline config/deploy/production.rb:17]: せやね
コメントを入れておいてもいいかも

Comment (ogidow) [inline config/deploy/production.rb:17]: タスクとして残しておきました！

Comment (shimoju) [inline config/deploy/production.rb:17]: あざます！

**Comment by shimoju** [comment]
LGTM 🆗

**Comment by ogidow** [comment]
やった！マージします！！

**Pull request merged**